### PR TITLE
adds namespace yaml

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2737.

### CODE REVIEW

File: helm/e2e-app-chart/templates/deployment.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: extensions/v1beta1
 kind: Deployment
 metadata:
   name: e2e-app
-  namespace: e2e
+  namespace: e2e-app
```
Comment (xh3b4sd) [inline helm/e2e-app-chart/templates/deployment.yaml:5]: Sorting the namespace.

File: helm/e2e-app-chart/templates/namespace.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: e2e-app
```
Comment (xh3b4sd) [inline helm/e2e-app-chart/templates/namespace.yaml:4]: Here is the new namespace.

File: helm/e2e-app-chart/templates/deployment.yaml [hunk 3]
```diff
@@ -15,7 +15,16 @@ spec:
...
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                  - key: "app"
+                    operator: In
+                    values:
+                    - e2e-app
+              topologyKey: "kubernetes.io/hostname"
```
Comment (xh3b4sd) [inline helm/e2e-app-chart/templates/deployment.yaml:27]: Adding anti affinity. We want to run a "HA" setup to ensure requests are not dropped even if nodes scale/fail.

**Review by xh3b4sd — COMMENTED** [review]
I tested this on minikube so far. Towards anti affinity one of the two pods is in mending so I assume this works because minikube only has one node. One issue left is how to request the e2e-app from within our build containers but this will be sorted separately.

**Review by MarcelMue — APPROVED** [review]
This is LGTM.

**Pull request merged**